components/Input/RoutePicker/RouterPickerWalletConnect.tsx

```diff
@@ -68,10 +80,17 @@ const PickerWalletConnect: FC = () => {
     </>
 }
 
-const WalletButton: FC<{ wallets: Wallet[], onOpenModalClick: () => void }> = ({ wallets, onOpenModalClick }) => {
+const WalletButton: FC<{ wallets: Wallet[], pickerSelectedWallets: ReturnType<typeof usePickerSelectedWalletStore>['pickerSelectedWallets'], onOpenModalClick: () => void }> = ({ wallets, onOpenModalClick, pickerSelectedWallets }) => {
 
     const wallet = wallets[0]
 
+    wallets.forEach(w => {
+        const selectedWallet = pickerSelectedWallets?.find(sw => sw.provider === w.providerName)
+        if (selectedWallet && selectedWallet.address && w.address !== selectedWallet.address) {
+            w.address = selectedWallet.address
```

Comment:
We should not mutate props. You can have a new array of wallets that is mapped from passed wallets